Cabal/src/Distribution/Simple/Haddock.hs

```diff
@@ -395,26 +395,28 @@ haddock pkg_descr lbi suffixes flags' = do
 
             return $ PackageIndex.insert ipi index
       CFLib flib ->
-        ( when (flag haddockForeignLibs) $ do
-            withTempDirectoryEx verbosity tmpFileOpts (buildDir lbi') "tmp" $
-              \tmp -> do
-                smsg
-                flibArgs <-
-                  fromForeignLib
-                    verbosity
-                    tmp
-                    lbi'
-                    clbi
-                    htmlTemplate
-                    version
-                    flib
-                let libArgs' = commonArgs `mappend` flibArgs
-                runHaddock verbosity tmpFileOpts comp platform haddockProg True libArgs'
-        )
+        when
+          (flag haddockForeignLibs)
+          ( do
+              withTempDirectoryEx verbosity tmpFileOpts (buildDir lbi') "tmp" $
+                \tmp -> do
+                  smsg
+                  flibArgs <-
```

Comment:
I do not fancy this one

Reply:
I get that. I looked at it a while and concluded the difficulty in reading the code block was there beforehand.

Reply:
I agree, perhaps you should consider trying to make this part it more readable.

Reply:
Personally, I'm a heavy user of `$` and I also don't like these new parens. But I can live with them no problem.